client/state/first-view/selectors.js

```diff
@@ -1,9 +1,19 @@
+import { FIRST_VIEW_START_DATES } from './constants';
+
+export function doesViewHaveFirstView( view ) {
+	return !! ( FIRST_VIEW_START_DATES[ view ] );
+}
+
 export function isViewVisible( state, view ) {
-	return ( ! isViewDisabled( state, view ) ) && -1 !== state.firstView.visible.indexOf( view );
+	return (
+		doesViewHaveFirstView( view ) &&
```

Comment:
Since `isViewEnabled` also makes this check and is called from here, you can just get rid of the `doesViewHaveFirstView` check here.